source/httpfs/migrator.go

```diff
@@ -0,0 +1,137 @@
+package httpfs
+
+import (
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/golang-migrate/migrate/v4/source"
+)
+
+// Migrator is a helper service for creating new source drivers working with
+// http.FileSystem instances. It implements all source.Driver interface methods
+// except for Open(). New driver could embed this struct and add missing Open()
+// method.
+//
+// To prepare Migrator for use Init() function.
+type Migrator struct {
```

Comment:
Why not call this `HTTPFS`? `Migrator` is a misleading name since this doesn't actually do any migrations.

Reply:
Resolving in favor of this thread: https://github.com/golang-migrate/migrate/pull/293#discussion_r357451371